File: 3120-count-the-number-of-special-characters-i/3120-count-the-number-of-special-characters-i.cpp
class Solution {
public:
    int numberOfSpecialChars(string word) {
        int cnt = 0;
        unordered_map<char, int>mp1, mp2;
        // vector<char>v1,v2;
        for(int i=0; i<word.size(); i++){
            if(word[i] >= 'a' && word[i] <= 'z')mp1[word[i]]++;
            if(word[i] >= 'A' && word[i] <= 'Z')mp2[word[i]]++;
        }
        for(auto it:mp1){
            for(auto its:mp2){
                if(it.first == tolower(its.first))cnt++;
            }
        }
        return cnt;
    }
};

File: 3120-count-the-number-of-special-characters-i/3120-count-the-number-of-special-characters-i.java
class Solution {
    public int numberOfSpecialChars(String word) {
        int ans=0;
        HashSet<Character> hs=new HashSet<>();
        for(int i=0;i<word.length();i++){
            if(Character.isUpperCase(word.charAt(i))) hs.add(word.charAt(i));
        }
        for(int i=0;i<word.length();i++){
            if(hs.contains((char)(word.charAt(i)-32))) {
                hs.remove((char)(word.charAt(i)-32));
                // System.out.println(i);
                ans++;
            }
            
        }
        return ans;
    }
}